### PR TITLE
feature/CP-6864-android-app-banners-targeting-by-events-from-previous-sessions-v1

### PR DESCRIPTION
**Pull request by unnaticleverpush**
Added feature for app banner target track event

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/database/TableBannerTrackEventDao.java [hunk 1]
```diff
@@ -19,20 +19,20 @@ public interface TableBannerTrackEventDao {
   @Update
   void update(TableBannerTrackEvent bannerTrackEvent);
 
-  @Query("SELECT * FROM TableBannerTrackEvent WHERE bannerId = :bannerId AND trackEventId = :trackEventId "
+  @Query("SELECT * FROM cleverpush_table_banner_track_event WHERE bannerId = :bannerId AND trackEventId = :trackEventId "
           + "AND property = :property AND value = :value AND relation = :relation AND fromValue = :fromValue AND toValue = :toValue")
   List<TableBannerTrackEvent> getBannerTrackEvent(String bannerId, String trackEventId, String property, String value, String relation, String fromValue, String toValue);
 
-  @Query("SELECT * FROM TableBannerTrackEvent WHERE trackEventId = :trackEventId ")
+  @Query("SELECT * FROM cleverpush_table_banner_track_event WHERE trackEventId = :trackEventId ")
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/database/TableBannerTrackEventDao.java:26]: trackEventId -> eventId

File: cleverpush/src/main/java/com/cleverpush/database/TableBannerTrackEventDao.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
 @Dao
 public interface TableBannerTrackEventDao {
 
-  @Query("SELECT * FROM TableBannerTrackEvent ORDER BY Id ASC")
+  @Query("SELECT * FROM cleverpush_table_banner_track_event ORDER BY Id ASC")
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/database/TableBannerTrackEventDao.java:13]: cleverpush_tracked_events

File: cleverpush/src/main/java/com/cleverpush/database/TableBannerTrackEventDao.java [hunk 3]
```diff
@@ -19,20 +19,20 @@ public interface TableBannerTrackEventDao {
   @Update
   void update(TableBannerTrackEvent bannerTrackEvent);
 
-  @Query("SELECT * FROM TableBannerTrackEvent WHERE bannerId = :bannerId AND trackEventId = :trackEventId "
+  @Query("SELECT * FROM cleverpush_table_banner_track_event WHERE bannerId = :bannerId AND trackEventId = :trackEventId "
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/database/TableBannerTrackEventDao.java:22]: Do we need `bannerId` in the table? Shouldn't the list of tracked events + properties be enough?

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/database/TableBannerTrackEventDao.java (hunk 3 above)
Comment (unnaticleverpush) [reply cleverpush/src/main/java/com/cleverpush/database/TableBannerTrackEventDao.java:22]: Yes, we need a banner ID for the count. I have 2 banners with the same event ID same property, and a different count than we need banner ID
for 1st banner, the past 3 days for the second banner past 5 days.

Screenshot 2023-11-28 at 8 00 34 PM

Screenshot 2023-11-28 at 8 01 54 PM

**Comment by gbhrdt** [comment]
Resolve conflicts @unnaticleverpush

**Comment by unnaticleverpush** [comment]
Conflict is resolved @gbhrdt

**Pull request merged**